packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -6009,12 +6013,20 @@ namespace Tpetra {
     // without unpacking the import/export buffer, we don't know how many of the
     // numPacketsPerLID[i] LIDs exist in the target. Below, it is assumed that
     // none do, and padding is requested for all.
-    for (size_t i = 0; i < numEnt; ++i) {
-      auto result = padding.insert (importLIDs_h[i], numPacketsPerLID_h[i]);
+    //
+    // Use tmp_padding since Kokkos::UnorderedMap does not allow re-insertion
+    std::map<local_ordinal_type, size_t> tmp_padding;
```

Comment:
@mhoemmen - this is the only functional change to the code (the rest being some indentation and extra comments to debug/failure messages).

The case that Albany was exercising had repeated indices in the `importLIDs`, since `Kokkos::UnorderedMap` does not allow re-insertion, use `std::map` to properly accumulate the extra capacity needed and then pass to `Kokkos::UnorderedMap`

Reply:
I'm pretty sure that `Kokkos::UnorderedMap` allows reinsertion.  You just have to check the return value of `insert`; it should say whether the thing is in the map already.  On the other hand, if this works, I'll be happy to approve :-) .  We can always go back later and fix this by using `UnorderedMap::insert`'s return value.